craft_providers/bases/buildd.py

```diff
@@ -271,6 +266,32 @@ def setup(
         self._setup_apt(executor=executor, deadline=deadline)
         self._setup_snapd(executor=executor, deadline=deadline)
 
+    def _disable_automatic_apt(
+        self, *, executor: Executor, deadline: Optional[float]
+    ) -> None:
+        """Disable automatic apt actions.
+
+        This should happen as soon as possible in the instance overall setup, to reduce the
+        chances of an automatic apt work being triggered during the setup itself (because it
+        includes apt work which may clash the triggered unattended jobs).
+
+        :param executor: Executor for target container.
+        :param deadline: Optional time.time() deadline.
+        """
+        _check_deadline(deadline)
+        # set the verification frequency in 10000 days and disable the upgrade
+        content = dedent(
+            """\
+            APT::Periodic::Update-Package-Lists "10000";
```

Comment:
Doing this now would mean that me need to add craft-parts support to update the cache before fetching new things.